fastestimator/backend/tensor_normalize.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2022 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import Tuple, TypeVar, Union, List
+
+import torch
+import numpy as np
+import tensorflow as tf
+
+
+from fastestimator.backend.reduce_mean import reduce_mean
+from fastestimator.backend.reduce_std import reduce_std
+from fastestimator.backend.to_tensor import to_tensor
+
+Tensor = TypeVar('Tensor', tf.Tensor, torch.Tensor, np.ndarray)
+allowed_data_types = Union[None, float, Tuple[float, ...], List[float]]
+
+
+def normalize(tensor: Tensor, mean: allowed_data_types, std: allowed_data_types, epsilon: float = 1e-7) -> Tensor:
```

Comment:
breaking off allowed_data_types strikes me as not super useful to developer since there's no obvious intuition as to what it would be. Maybe bring back inline and just replace Tuple[float] List[float] with Iterable[float]